addons/mail/models/mail_channel.py

```diff
@@ -256,6 +256,15 @@ def write(self, vals):
                 ('res_id', 'in', self.ids)
             ])._moderate_accept()
 
+        # Reflect changes when switching a channel to a mailing one or vice versa
+        if 'email_send' in vals:
+            notifications = []
+            for channel in self:
+                notifications.append([(self._cr.dbname, 'mail.channel', channel.id), {
+                    'info': 'toggle_mass_mailing',
+                }])
```

Comment:
if `channel.email_send == vals.get('email_send')` it's pointless to send the notification

Reply:
@seb-odoo 
I am still unable to get your point because I have made these changes based on the below considerations. just let me know still I missed anything.

1) "email_send" key only comes in the write method when the value has been changed.
2) We have to notify the user in both the case if the value of the "email_send" is True or False.
   Ex: 
        1) User have marked True the "Send messages by email" checkbox >> vals.get('mass_mailing') is True >> Need to     reflect the chagnes to each member.
        2) Now User is reverting back the changes with uncheck the checkbox >> vals.get('mass_mailing') is False >> Also need to reflect the changes to each member again with removing the mailing related settings.

So I have handled both the cases using only checking ( 'email_send' in vals ) or not.

Reply:
Thank you, it's better now. What I meant is that you could save in a local variable the previous value for each channel before the write, and compare it to the given value to decide whether it changed or not, but it's not worth the effort. So ok for now for this remark.